Source/ACE.Server/WorldObjects/Player_Spells.cs

```diff
@@ -141,9 +141,9 @@ public void OnItemLevelUp(WorldObject item, int prevItemLevel)
 
             var levelDiff = prevItemLevel - (item.ItemLevel ?? 0);
 
-            var prevSpells = GetSpellSet((EquipmentSet)item.EquipmentSetId, setItems, levelDiff);
+            var prevSpells = GetSpellSet((EquipmentSet)item.EquipmentSetId, item.ItemXpStyle ?? 0, setItems, levelDiff);
```

Comment:
maybe just pass in item here, and have GetSpellSet fetch (EquipmentSet)Item.EquipmentSetId and item.ItemXpStyle ?? 0 in 1 place?

Reply:
or not sure if item is even needed, if it should just verify setItems.Count > 0, get the first item, and use the EquipmentSetId/ItemXpStyle from that? (would require the calling the function to ensure that the items in setItems match up)